File: Lesson9/data/src/main/java/ru/mirea/vasilev/data/repository/MovieRepositoryImpl.java
package ru.mirea.vasilev.data.repository;

import java.time.LocalDate;

import ru.mirea.vasilev.data.storage.MovieStorage;
import ru.mirea.vasilev.data.storage.models.Movie;
import ru.mirea.vasilev.domain.repository.MovieRepository;

public class MovieRepositoryImpl implements MovieRepository {
    private MovieStorage movieStorage;
    public MovieRepositoryImpl(MovieStorage movieStorage) {
        this.movieStorage = movieStorage;
    }

    @Override
    public boolean saveMovie(ru.mirea.vasilev.domain.models.Movie movie) {
        movieStorage.save(mapToStorage(movie));
        return true;
    }

    @Override
    public ru.mirea.vasilev.domain.models.Movie getMovie() {
        Movie movie = movieStorage.get();
        return mapToDomain(movie);
    }

    private Movie mapToStorage(ru.mirea.vasilev.domain.models.Movie movie){
        String name = movie.GetName();
        return new Movie(2, name, LocalDate.now().toString());
    }

    private ru.mirea.vasilev.domain.models.Movie mapToDomain(Movie movie){
        String name = movie.getName();
        return new ru.mirea.vasilev.domain.models.Movie(movie.getId(), name);
    }
}

File: lesson9/app/src/main/java/ru/mirea/vasilevmn/lesson9/data/repository/MovieRepositoryImpl.kt
package ru.mirea.vasilevmn.lesson9.data.repository

import android.content.Context
import android.content.SharedPreferences
import ru.mirea.vasilevmn.lesson9.domain.models.Movie
import ru.mirea.vasilevmn.lesson9.domain.repository.MovieRepository


class MovieRepositoryImpl(context: Context) : MovieRepository {

    private val sharedPreferences: SharedPreferences =
        context.getSharedPreferences("FavoriteMovies", Context.MODE_PRIVATE)

    override fun saveMovie(movie: Movie?): Boolean {
        return if (movie != null) {
            val editor = sharedPreferences.edit()
            editor.putInt("movie_id", movie.id)
            editor.putString("movie_name", movie.name)
            editor.apply()
            true
        } else {
            false
        }
    }

    override val movie: Movie?
        get() {
            val id = sharedPreferences.getInt("movie_id", -1)
            val name = sharedPreferences.getString("movie_name", null)
            return if (id != -1 && name != null) {
                Movie(id, name)
            } else {
                null
            }
        }
}